generate_response.py

```diff
@@ -817,6 +821,12 @@ def generate_response(model, msgs, topn, temperature, args, open_source_model, t
         else:
             response_list.append(communicator_response)    
         return response_list  
+    elif model == 'AgentCoder':
+        responses = programmer_main(model, "python", msgs, openai.api_key)
```

Comment:
`msgs` is a list IIUC, is this same as `dataset` in `programmer_main`? please advice

Reply:
have you tested this code? I don't think this works, as `dataset` argument is a path in `programmer_main`.

Reply:
Due to the requirement of "entry_point" in the AgentFramework, I have added functionality that loads HumanEvalComm.json inside the AgentCoder block. We send the prompt and the entry_point as a list of dictionaries. 

Only addition required is for the start and end index. Will do so in the next (and hopefully final) commit.